src/main/java/com/example/spring/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,134 @@
+package com.example.spring.service.order;
+
+import com.example.spring.dto.order.OrderDto;
+import com.example.spring.dto.order.OrderItemDto;
+import com.example.spring.dto.order.OrderItemRequestDto;
+import com.example.spring.dto.order.UpdateOrderStatusRequestDto;
+import com.example.spring.exception.EntityNotFoundException;
+import com.example.spring.mapper.OrderItemMapper;
+import com.example.spring.mapper.OrderMapper;
+import com.example.spring.model.CartItem;
+import com.example.spring.model.Order;
+import com.example.spring.model.OrderItem;
+import com.example.spring.model.ShoppingCart;
+import com.example.spring.model.User;
+import com.example.spring.repository.order.OrderItemRepository;
+import com.example.spring.repository.order.OrderRepository;
+import com.example.spring.service.cart.CartService;
+import com.example.spring.service.user.UserService;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final UserService userService;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final CartService cartService;
+    private final OrderItemMapper orderItemMapper;
+
+    @Transactional
+    @Override
+    public OrderDto addOrder(Long userId, OrderItemRequestDto orderItemRequestDto) {
+        Order order = createOrder(userId, orderItemRequestDto);
+        orderRepository.save(order);

```

Comment:
Also, instead of createOrder method, you can use 
`Order order = orderMapper.cartToOrder(cart, orderDto.shippingAddress());`

where
```
@Mapping(target = "id", ignore = true)
@Mapping(target = "total", source = "cart.cartItems", qualifiedByName = "total")
@Mapping(target = "orderItems", source = "cart.cartItems")
Order cartToOrder(ShoppingCart cart, String shippingAddress);
```

and 
```
@Named("total")
    default BigDecimal getTotal(Set<CartItem> cartItems) {
        return cartItems.stream()
                .map(i -> i.getBook().getPrice().multiply(BigDecimal.valueOf(i.getQuantity())))
                .reduce(BigDecimal.ZERO, BigDecimal::add);
    }
```

Reply:
I had different hibernate mistakes so i decided to leave createOrder method

Reply:
Exceptions doesn't mean that you need to complicate your code) 
You can ask in QnA how to solve them and make code more readable and simplier